test/parallel/test-cluster-worker-disconnect.js

```diff
@@ -8,6 +10,11 @@ if (cluster.isWorker) {
 
   }).listen(common.PORT, '127.0.0.1');
 
+  // Save flag-file on disconnect
+  cluster.worker.on('disconnect', function(){
+    fs.writeFileSync(path.join(__dirname, process.pid + ''), '');
```

Comment:
Instead of doing this, can you check out `test/parallel/test-cluster-worker-death.js`, and do something similar.